src/components/SelectedData/SelectedDataContextMenu/index.js

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useRef } from 'react';
+
+import PropTypes from 'prop-types';
+
+import {
+  selectedDataContextMenu,
+  selectedDataContextMenuClose,
+} from './SelectedDataContextMenu.module.scss';
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+  faEdit,
+  faMinusCircle,
+  faTrash,
+  faPlus,
+  faTimes,
+} from '@fortawesome/pro-solid-svg-icons';
+
+const SelectedDataContextMenu = ({
+  closeAction,
+  addAction,
+  deleteAllAction,
+  deselectAllAction,
+}) => {
+  const containerRef = useRef();
+
+  const handleClick = e => {
+    const _Target = e.target;
+
+    if (!containerRef.current) return;
+
+    if (!containerRef.current.contains(_Target)) {
+      closeAction();
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('click', handleClick);
+
+    return () => {
+      document.removeEventListener('click', handleClick);
+    };
+  }, []);
```

Comment:
I'd usually create a customHook for this but since the lack of time its okay

Reply:
I think I have one living in another project, I will port it over if I find myself using it in multiple places.